tools/topology/sof-apl-keyword-detect.m4

```diff
@@ -0,0 +1,102 @@
+#
+# Topology for ApolloLake with direct attach digital microphones array for
+# keyword detection and triggering use case.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM 0 <-------+- KPBM 0 <-- B0 <-- DMIC6 (DMIC01)
+#               |
+# Keyword <-----+
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     frames, deadline, priority, core)
+
+# Passthrough capture pipeline 1 on PCM 0 using max 2 channels.
+# Schedule 16 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-kfbm-capture.m4,
+	1, 0, 2, s32le,
+	16, 1000, 0, 0)
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     frames, deadline, priority, core)
+
+
+# capture DAI is DMIC 0 using 1000 periods (64kB buffer for 2 secs of audio)
+# Buffers use s16le format, with 16 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	1, DMIC, 0, NoCodec-6,
+	PIPELINE_SINK_1, 1000, s16le,
+	16, 1000, 0, 0)
+
+dnl PCM_CAPTURE_ADD(name, pipeline, capture)
+PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_1)
+
+# keyword detector pipe
+dnl PIPELINE_ADD(pipeline,
+dnl     pipe id, max channels, format,
+dnl     frames, deadline, priority, core)
+PIPELINE_ADD(sof/pipe-detect.m4, 2, 2, s16le, 16, 1000, 0, 0, PIPELINE_SCHED_COMP_1)
```

Comment:
320 frames for 20 ms periods.

Reply:
@keyonjie the frame size and deadline should be defined in the top level m4 and be common for all pipelines in the topology.

Reply:
@ranj063 @keyonjie Only for keyword detector related pipelines. All other capture and render streams should operate on normal 1 ms frames.

Reply:
@tlauda I'd assume that both the pipelines in sof-apl-keyword-detect.m4 would use the same frame size and deadline. Or do you mean to say only the slector->detect pipeline should use this?

Reply:
ditto.

Reply:
@ranj063 @tlauda Yes, for both should be 20 ms.

Reply:
@tlauda so actually all buffers in this tplg file need to be 20ms.

And with 1 period or 2?

Reply:
@keyonjie 2 unless we confirm that 1 also works.

Reply:
OK, let me update it like that.